lib/flex_algo.c

```diff
@@ -17,20 +17,27 @@
 
 #include "flex_algo.h"
 
+DEFINE_MTYPE_STATIC(LIB, FLEX_ALGOS, "Flex-Algo Allocator / Releaser");
 DEFINE_MTYPE_STATIC(LIB, FLEX_ALGO, "Flex-Algo Definition");
 
 struct flex_algos *flex_algos_alloc(flex_algo_allocator_t allocator,
 				    flex_algo_releaser_t releaser)
 {
 	struct flex_algos *flex_algos;
 
-	flex_algos = XCALLOC(MTYPE_FLEX_ALGO, sizeof(*flex_algos));
+	flex_algos = XCALLOC(MTYPE_FLEX_ALGOS, sizeof(*flex_algos));
 	flex_algos->flex_algos = list_new();
```

Comment:
better to use `sizeof(struct flex_algos)`

Reply:
both are working
Out of scope of the current pull-request. I am doing a separate PR for that

Reply:
done